routes/auth.js

```diff
@@ -0,0 +1,76 @@
+const express = require('express');
+const bcrypt = require('bcrypt');
+const createToken = require('../utils/createToken');
+const { models } = require('../models/index');
+
+const routes = express.Router();
+
+routes.post(
+  '/register',
+  // add validation middleware
+  async (req, res, next) => {
+    try {
+      // Check if this name already exist in DB
+      let { name, hotel_id, password } = req.body;
+      const [existingUser] = await models.User.where({ name });
+
+      if (existingUser) {
+        return res.status(400).json({
+          message: 'That name is taken already.',
+        });
+      }
+
+      password = bcrypt.hashSync(password, 10);
+
+      // add new user to the DB, rewrite the password to be the hashed pw
+      const user = await models.User.create({ ...req.body, password });
+
+      // remove password from the returned user object, so it's not sent to FE
+      user.password = undefined;
+
+      // generate token
+      const token = createToken({
+        id: user.id,
+        name,
+        hotel_id,
+        chat_id: user.chat_id,
+      });
+
+      // send the user info and token in response
+      res.status(201).json({ user, token });
+    } catch (err) {
+      next(err);
+    }
+  }
+);
+
+routes.post(
```

Comment:
If we have different endpoints for staff and guest auth, should we reserve /login for guests, it being the cleanest one?

Reply:
The path, which user sees will be set on fronted with React Router.

Reply:
**finds way to delete hideous PR comment**